internal/mode/static/state/graph/graph.go

```diff
@@ -293,3 +299,46 @@ func gatewayExists(
 
 	return exists
 }
+
+// PlusSecretFileType describes the type of Secret file used for NGINX Plus.
+type PlusSecretFileType int
+
+const (
+	// PlusReportJWTToken is the file for the NGINX Plus JWT Token.
+	PlusReportJWTToken PlusSecretFileType = iota
+	// PlusReportCACertificate is the file for the NGINX Instance Manager CA certificate.
+	PlusReportCACertificate
+	// PlusReportClientSSLCertificate is the file for the NGINX Instance Manager client certificate.
+	PlusReportClientSSLCertificate
+	// PlusReportClientSSLKey is the file for the NGINX Instance Manager client key.
+	PlusReportClientSSLKey
+)
+
+// PlusSecretFile specifies the type and content of an NGINX Plus Secret file.
+// A user provides the names of the various Secrets on startup, and we store this info in a map to cross-reference with
+// the actual Secrets that exist in k8s.
+type PlusSecretFile struct {
+	// FieldName is the field name within the Secret that holds the data for this file.
+	FieldName string
+	// Content is the content of this file.
+	Content []byte
+	// Type is the type of Secret file.
+	Type PlusSecretFileType
+}
+
+// setPlusSecretContent finds the k8s Secret object associated with a PlusSecretFile object, and sets its contents.
+func setPlusSecretContent(
+	clusterSecrets map[types.NamespacedName]*v1.Secret,
+	plusSecrets map[types.NamespacedName][]PlusSecretFile,
+) map[types.NamespacedName][]PlusSecretFile {
+	for name, plusSecretFiles := range plusSecrets {
+		if secret, ok := clusterSecrets[name]; ok {
+			for idx, file := range plusSecretFiles {
+				file.Content = secret.Data[file.FieldName]
+				plusSecrets[name][idx] = file
+			}
+		}
+	}
+
+	return plusSecrets
```

Comment:
It's confusing to me that we modify the map in place and then return that map. 

We should do one or the other and be clear about it.